dataclients/kubernetes/ingress.go

```diff
@@ -128,7 +128,11 @@ func setPath(m PathMode, r *eskip.Route, p string) {
 	case PathRegexp:
 		r.PathRegexps = []string{p}
 	default:
-		r.PathRegexps = []string{"^" + p}
+		if p == "/" {
```

Comment:
What if p is empty?

Reply:
Well before it was "^" + p so I guess weirdest hypothetical case is when p == "" which would go to "^()" which in the regex world is equivalent to "^" so I don't think it will change behavior and will work ok. 

I had to add the p == "/" case today to fix some of the tests. Without that special case clause it seemed to be generating additional "catchall" cases in the route tables. The is coz the hasCatchAllRoutes(routes []*eskip.Route) looks for "^/" (see https://github.com/zalando/skipper/blob/dd70bd65e7f99cfb5dd6b6f71885d9fe3b2707f6/dataclients/kubernetes/ingress.go#L746).